### ISSUE TITLE
KeyError: "[E002] Can't find factory for 'neuralcoref' in Ubuntu 16.04 Python 3 when trying to load a downloaded model

### ISSUE THREAD

**Issue by rajeshkumargp** [huggingface/neuralcoref#81]
Hi,  I am using Ubuntu 16.04 ,Python 3. 
I have downloaded https://github.com/huggingface/neuralcoref-models/releases/download/en_coref_md-3.0.0/en_coref_md-3.0.0.tar.gz and then extracted it. 

```
import spacy
mpath='/home/user/en_coref_md-3.0.0/en_coref_md/en_coref_md-3.0.0'
#  Contents of the directory
#  meta.json  ner  neuralcoref  parser  tagger  tokenizer  vocab
nlp=spacy.load(mpath)

```
Getting below error message: 

`
KeyError: "[E002] Can't find factory for 'neuralcoref'. This usually happens when spaCy calls `nlp.create_pipe` with a component name that's not built in - for example, when constructing the pipeline from a model's meta.json. If you're using a custom component, you can write to `Language.factories['neuralcoref']` or remove it from the model meta and add it via `nlp.add_pipe` instead."
`

Please suggest/guide me to fix this issue.

**Comment by tanzf** [comment]
I have encountered the same problem, but I don't know how to modify it. Have you solved it? Can you teach me?

**Comment by oats-meal** [comment]
I'm getting this error as well. I'm running neuralcoref 3.1 in Python3.6 and also using PyCharm.

**Comment by thomwolf** [comment]
can you try to pip install the downloaded model instead of extracting it?

**Comment by rbhambriiit** [comment]
![image](https://user-images.githubusercontent.com/13808100/50191868-a3e7dc00-0327-11e9-95f5-38972961e6bb.png)


Successfully built en-coref-md
Installing collected packages: en-coref-md
Successfully installed en-coref-md-3.0.0



![image](https://user-images.githubusercontent.com/13808100/50191877-be21ba00-0327-11e9-9994-9f182441f0c4.png)

OSError: [E050] Can't find model 'en_coref_md'. It doesn't seem to be a shortcut link, a Python package or a valid path to a data directory.

![image](https://user-images.githubusercontent.com/13808100/50191885-c974e580-0327-11e9-9488-1f97c919a212.png)

ModuleNotFoundError: No module named 'en_coref_md'

nlp = spacy.load('en-coref-md') -> also failed.

**Comment by ewengillies** [comment]
So I've done some terrible things, but I managed to hack around this. **Note** if you can, *please* install it via pip the way the creators intended.  The models worked flawlessly for me when I did this.  

I have a very special use case that benefits from dynamically loading the different en_coref_XX models from local directories.  

Now into the dangerzone. If you look at the `__init__.py` for the coreference models, you will find:

```python
# coding: utf8
from __future__ import unicode_literals

from pathlib import Path
from spacy.util import load_model_from_init_py, get_model_meta
from en_coref_sm.neuralcoref import NeuralCoref

__version__ = get_model_meta(Path(__file__).parent)['version']


def load(**overrides):
    disable = overrides.get('disable', [])
    overrides['disable'] = disable + ['neuralcoref']
    nlp = load_model_from_init_py(__file__, **overrides)
    coref = NeuralCoref(nlp.vocab)
    coref.from_disk(nlp.path / 'neuralcoref')
    nlp.add_pipe(coref, name='neuralcoref')
    return nlp
```

I don't know a huge amount about spacy and custom spacy models, but it stands to reason that if you install this properly via pip, the above code is run and everyone is happy.  The `load()` function removes `neuralcoref` from the model (which is detected in the `meta.json` file), loads the model from the `__init__.py` so the rest of the components named in the `meta.json` are added to the pipeline, then `NeuralCoref` is added to the pipeline (again, not an expert, 50% chance I missed something entirely).  

To load the model from  a path, one can recreate this `load()` function like this:

```python
import os
import sys
import spacy

# Our horrible hack of a function
def load_coref_model_dynamically(model_path, model_name="en_coref_sm"):                                                                                                                                                               
     # Find the __init__.py for the model
     model_path += "/{}/__init__.py".format(model_name)
     # Load the model, removing the coreference part
     nlp = spacy.util.load_model_from_init_py(model_path,                                                                                                                                                    
                                              disable=["neuralcoref")
     # My least favorite line, to tell python where to find the neuralcoref module that contains
     # NeuralCoref object
     sys.path.append(os.path.join(nlp.path, "..", "neuralcoref"))
     # Import our new best friend                                                                                                                                            
     import neuralcoref
     # Build the object and add it to the pipeline
     coref = neuralcoref.NeuralCoref(nlp.vocab)                                                                                                                                                              
     coref.from_disk(nlp.path / 'neuralcoref')                                                                                                                                                               
     nlp.add_pipe(coref, name="neuralcoref")                                                                                                                                                                 
     return nlp  
```

Again, don't do this unless you need to.  My *single* test case passes, so that's what I'm going on when I say this "works". 

`huggingface`: is there any chance that we would be able to import these models from a path without first installing them with `pip`?  Thanks!

**Comment by thomwolf** [comment]
This should be fixed in the new release (4.0) which is on PyPI.
Please check the README for details.
Fell free to open a new issue if you still experience installation problems.
